configure.ac

```diff
@@ -362,10 +362,14 @@ AC_SUBST(LIBAUDIT)
 AC_CHECK_HEADERS(crypt.h)
 
 BACKUP_LIBS=$LIBS
-AC_SEARCH_LIBS([crypt],[crypt])
-case "$ac_cv_search_crypt" in
-	-l*) LIBCRYPT="$ac_cv_search_crypt" ;;
-	*) LIBCRYPT="" ;;
+LIBCRYPT=""
+AC_SEARCH_LIBS([crypt_r],[crypt])
+case "$ac_cv_search_crypt_r" in
+	-l*) LIBCRYPT="$ac_cv_search_crypt_r" ;;
+	no) AC_SEARCH_LIBS([crypt],[crypt])
+	case "$ac_cv_search_crypt" in
+		-l*) LIBCRYPT="$ac_cv_search_crypt" ;;
+	esac ;;
```

Comment:
Please re-introduce the check for `crypt_gensalt_rn` for the same reason: to fix potentially broken builds when libc provides `crypt_r` but doesn't provide `crypt_gensalt_rn`.